server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -124,6 +124,15 @@ private static void internalParseDocument(MetadataFieldMapper[] metadataFieldsMa
 
             if (context.root().isEnabled() == false) {
                 // entire type is disabled
+                if (context.isSourceSynthetic()) {
+                    context.addIgnoredField(
+                        new IgnoredSourceFieldMapper.NameValue(
+                            MapperService.SINGLE_MAPPING_NAME,
+                            0,
+                            XContentDataHelper.encodeToken(context.parser())
+                        )
+                    );
+                }
                 context.parser().skipChildren();
```

Comment:
should `context.parser().skipChildren()` only be executed if `context.isSourceSynthetic() == false`? If so then add an `else` clause here?

Reply:
I think it's fine as is, the `_ignored_source` entry created above includes all children so there's no point to further inspect this object. The tests support that this is the case?

Reply:
I see, this is a no-op for the new path.